cl/opinion_page/views.py

```diff
@@ -264,7 +265,14 @@ def view_docket(request: HttpRequest, pk: int, slug: str) -> HttpResponse:
     de_list = docket.docket_entries.all().prefetch_related("recap_documents")
     form = DocketEntryFilterForm(request.GET)
     if form.is_valid():
+        user: UserProfile.user = request.user
+        desc = user.is_authenticated and user.profile.sort_desc
         cd = form.cleaned_data
+        if desc and not cd.get("order_by"):
+            cd["order_by"] = DocketEntryFilterForm.DESCENDING
+            form = DocketEntryFilterForm(cd)
+            form.is_valid()
+            cd = form.cleaned_data
```

Comment:
I think this is a bit hacky, because we're creating and validating the form two times. There are a couple ways to work around this. 

1. Set the ordering before making the form.
2. Set the ordering within the form using a `clean_x` function.

I think number one is a so-so approach, but it suffers because request.GET is immutable, so to do it, you have to make a copy of request.GET and then fiddle with that.

I think the second option is a bit better, and the trick for it, I believe, is to pass the request as a second parameter to a custom `__init__` method on the `DocketEntryFilterForm`. This pattern is described here: 

https://stackoverflow.com/a/1057640/64911

If you do that, you'll have the `request` object in the form, and you can do a `clean_order_by` function that takes it into account and returns a nice value. This approach seems nicer to me too than the others, because it moves this fiddly logic out of the view and into the form, where it's hidden away.

I get the sense that you're trying to bang this out, and probably don't want to refactor this, but if you don't mind attempting this approach (assuming you agree with it generally), that'd be really great.

Reply:
I like to program in what I call "railroad siding" style.  Whenever there is a logical track to follow and something is outside of that track that creates a "siding" you want to rejoin the main line as soon as you can so no other logic needs to change. 

I like the idea of just creating or updating `clean_order_by` to handle the process.  I remember it being painful to get `request` into the clean part but it can be done.  Nice catch